src/datasources/account/account.datasource.ts

```diff
@@ -100,116 +157,153 @@ export class AccountDataSource implements IAccountDataSource {
     signer: string;
     code: string;
     codeGenerationDate: Date;
-  }): Promise<void> {
-    const [account] = await this.sql<Account[]>`UPDATE accounts
-                                                SET verification_code              = ${args.code},
-                                                    verification_code_generated_on = ${args.codeGenerationDate}
-                                                WHERE chain_id = ${args.chainId}
-                                                  AND safe_address = ${args.safeAddress}
-                                                  AND signer = ${args.signer}
-                                                RETURNING *`;
+  }): Promise<DomainVerificationCode> {
+    const [verificationCode] = await this.sql<VerificationCode[]>`
+        INSERT INTO verification_codes (account_id, code, generated_on)
+        SELECT id, ${args.code}, ${args.codeGenerationDate}
+        FROM accounts
+        WHERE chain_id = ${args.chainId}
+          AND safe_address = ${args.safeAddress}
+          AND signer = ${args.signer}
+          AND verified = false
+        ON CONFLICT (account_id)
+            DO UPDATE SET code         = ${args.code},
+                          generated_on = ${args.codeGenerationDate}
+        RETURNING *
+    `;
```

Comment:
Since the Verification Code is now separate from the Accounts, setting a verification code now does an "upsert" if the account is not verified.